### PR TITLE
Add a README with CI and Hackage badges

### PR DESCRIPTION
**Pull request by Kleidukos**
This PR adds a README file with the description of the project, as well as indicators of the CI status and the Hackage version.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# zlib [![CI][CI badge]][CI page] [![Hackage][Hackage badge]][Hackage page]
+
+> Compression and decompression in the gzip and zlib formats
```
Comment (emilypi) [inline README.md:3]:
```suggestion
**Compression and decompression in the gzip and zlib formats**
```

Comment (emilypi) [reply README.md:3]: I think this should render a little nicer

**Comment by Kleidukos** [comment]
@emilypi Considering that Travis.org closes shop soon, should I remove the configuration file from the repo, since we have GH Actions now?

File: README.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
+# zlib [![CI][CI badge]][CI page] [![Hackage][Hackage badge]][Hackage page]
```
Comment (Bodigrim) [inline README.md:1]: Given that CIs come and go, and `README` is frozen forever on Hackage, I wonder whether we want to add CI badges. Moreover, this particular link will rot away once we change the name of GHA workflow (e. g., switching from `haskell-ci`-generated to `haskell/actions/setup`-based).

File: README.md [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+**Compression and decompression in the gzip and zlib format**
+
+This package provides a pure interface for compressing and decompressing streams of data represented as lazy ByteStrings.
+It uses the zlib C library so it has high performance. It supports the "zlib", "gzip" and "raw" compression formats.
```
Comment (Bodigrim) [inline README.md:6]:
```suggestion
This package provides a pure interface for compressing and decompressing streams of data represented as lazy `ByteString`s.
It uses the `zlib` C library so it has high performance. It supports the `zlib`, `gzip` and `raw` compression formats.
```
Trivial changes attract most suggestions ;)

**Comment by Bodigrim** [comment]
> Considering that Travis.org closes shop soon, should I remove the configuration file from the repo

Sounds reasonable, Travis is not coming back.
(I actually tried to negotiate an open-source quota under new Travis rules, but was ignored)

File: README.md (hunk 3 above)
Comment (Kleidukos) [reply README.md:6]: I literally picked this up from the hackage homepage from zlib ;) Should I change that in the Cabal file as well?

File: README.md (hunk 2 above)
Comment (Kleidukos) [reply README.md:1]: The name is extracted from:

```
name: Haskell-CI
```
in the file. I don't understand what would prevent us from keeping it, if it's just something that we can set?

Comment (Bodigrim) [reply README.md:1]: Potentially we can, but this is one more thing to track and keep in mind. At some point we might end up with several GHA workflows. Shall we list all of them?.. What happens if the repo is moved under another organization?

Comment (Bodigrim) [reply README.md:1]: Another point is that it is weird when a badge in `README` for a fixed version on Hackage points to the last `HEAD` build. Failing `HEAD` build does not have any implications on the released version, but can cause such wrong impression.

This is not a big deal, if you do not feel convinced I'm not gonna press.

Comment (Kleidukos) [reply README.md:1]: I've given some thought to it, and considering that the README targets GitHub users first and foremost, and that Hackage builds the package itself, we don't really have to worry about this badge. But thanks for raising this point. :)

**Review by Bodigrim — APPROVED** [review]

**Review by emilypi — APPROVED** [review]
@Kleidukos agree with @Bodigrim here. Travis is gone, we github actions now.

**Pull request merged**